src/hooks/useUpdates.js

```diff
@@ -1,45 +1,39 @@
-import { useCallback } from 'react'
 import * as Updates from 'expo-updates'
 
-import useRefState from './useRefState'
-
 const useUpdates = () => {
-
-  // TODO: Once I upgrade to expo 50, get rid of this hook and use expo's Updates.useUpdates hook
-
-  const [ info, setInfo, getInfo ] = useRefState({
-    isUpdateAvailable: false,
-    isUpdatePending: false,
-    isChecking: true,
-    isDownloading: false,
-  })
-
-  const setUpdates = useCallback(
-    newInfo => {
-      setInfo({
-        ...getInfo(),
-        ...newInfo,
-      })
-    },
-    [],
-  )
-
-  const eventListener = event => {
-    setInfo({
-      isUpdateAvailable: event.type === Updates.UpdateEventType.UPDATE_AVAILABLE,
-      isUpdatePending: event.type === Updates.UpdateEventType.UPDATE_AVAILABLE,
-      isChecking: false,
-      isDownloading: false,
-    })
-  }
-
-  Updates.useUpdateEvents(eventListener)
+  // Use the official Updates.useUpdates() hook from Expo SDK 52
+  const {
+    currentlyRunning,
+    isUpdateAvailable,
+    isUpdatePending,
+    isChecking,
+    isDownloading,
+    availableUpdate,
+    downloadedUpdate,
+    checkError,
+    downloadError,
+    lastCheckForUpdateTimeSinceRestart,
+  } = Updates.useUpdates()
 
   return {
-    ...info,
-    setUpdates,
+    isUpdateAvailable,
+    isUpdatePending: isUpdatePending,
+    isChecking,
+    isDownloading,
+    // Legacy compatibility - map the new API to the old expected interface
+    setUpdates: () => {
```

Comment:
Given that this is deprecated and will now no longer do anything, should the references to this be removed, e.g. in App.js?

Are there any changes necessary for the uses of other values returned here?